drivers/sensor/ina23x/ina237.c

```diff
@@ -244,8 +244,8 @@ static int ina237_calibrate(const struct device *dev)
 	int ret;
 
 	/* see datasheet "Current and Power calculations" section */
-	val = (INA237_CAL_SCALING * config->current_lsb * config->rshunt) /
-	       10000000U;
+	val = (INA237_CAL_SCALING * ((config->current_lsb * config->rshunt) /
+	       1000U)) / 10000000U;
```

Comment:
This is a less accurate than it needs to be.   And again, it can be computed at build time with 64-bit math for free.

Reply:
Thanks for the input, I tried this with a few different methods, but note that it needs to be calcualted per each instance from the device tree, therefore making it required to be in the config structure. (please see ina230x.patch below)

I originally went down this road to try and get a  uint16_t .cal option created for each dt instance.
IE:
```c
.cal = INA230_CAL_SCALING * 1000000000U /	    \
			(DT_INST_PROP(inst, current_lsb_microamps) *\
			DT_INST_PROP(inst, rshunt_micro_ohms)),     \
```

I then realized that in real tests this method also had issues with precision and after changing it to:

ina230x.patch:
```
diff --git a/drivers/sensor/ina23x/ina230.c b/drivers/sensor/ina23x/ina230.c
index 6e7c23ba15..f3cc9486ae 100644
--- a/drivers/sensor/ina23x/ina230.c
+++ b/drivers/sensor/ina23x/ina230.c
@@ -187,14 +187,10 @@ static int ina230_attr_get(const struct device *dev, enum sensor_channel chan,
 static int ina230_calibrate(const struct device *dev)
 {
 	const struct ina230_config *config = dev->config;
-	uint16_t val;
 	int ret;
 
 	/* See datasheet "Programming" section */
-	val = (INA230_CAL_SCALING * 10000U) /
-	      ((config->current_lsb * config->rshunt) / 1000U);
-
-	ret = ina23x_reg_write(&config->bus, INA230_REG_CALIB, val);
+	ret = ina23x_reg_write(&config->bus, INA230_REG_CALIB, config->cal);
 	if (ret < 0) {
 		return ret;
 	}
@@ -276,7 +272,9 @@ static const struct sensor_driver_api ina230_driver_api = {
 		.bus = I2C_DT_SPEC_INST_GET(inst),		    \
 		.config = DT_INST_PROP(inst, config),		    \
 		.current_lsb = DT_INST_PROP(inst, current_lsb_microamps),\
-		.rshunt = DT_INST_PROP(inst, rshunt_micro_ohms),	    \
+		.rshunt = DT_INST_PROP(inst, rshunt_micro_ohms),    \
+		.cal = 5120000U / ((DT_INST_PROP(inst, current_lsb_microamps) *\
+			DT_INST_PROP(inst, rshunt_micro_ohms) / 1000U)),\
 		COND_CODE_1(DT_INST_NODE_HAS_PROP(inst, alert_gpios),\
 			    (INA230_CFG_IRQ(inst)), ())		    \
 	};							    \
diff --git a/drivers/sensor/ina23x/ina230.h b/drivers/sensor/ina23x/ina230.h
index 68a4bbef23..63955ca6ff 100644
--- a/drivers/sensor/ina23x/ina230.h
+++ b/drivers/sensor/ina23x/ina230.h
@@ -51,6 +51,7 @@ struct ina230_config {
 	uint16_t config;
 	uint32_t current_lsb;
 	uint32_t rshunt;
+	uint16_t cal;
 #ifdef CONFIG_INA230_TRIGGER
 	bool trig_enabled;
 	uint16_t mask;
diff --git a/drivers/sensor/ina23x/ina237.c b/drivers/sensor/ina23x/ina237.c
index 468f4af9f9..fe7485bb1d 100644
--- a/drivers/sensor/ina23x/ina237.c
+++ b/drivers/sensor/ina23x/ina237.c
@@ -244,10 +244,7 @@ static int ina237_calibrate(const struct device *dev)
 	int ret;
 
 	/* see datasheet "Current and Power calculations" section */
-	val = (INA237_CAL_SCALING * ((config->current_lsb * config->rshunt) /
-	       1000U)) / 10000000U;
-
-	ret = ina23x_reg_write(&config->bus, INA237_REG_CALIB, val);
+	ret = ina23x_reg_write(&config->bus, INA237_REG_CALIB, config->cal);
 	if (ret < 0) {
 		return ret;
 	}
@@ -379,7 +376,9 @@ static const struct sensor_driver_api ina237_driver_api = {
 		.config = DT_INST_PROP(inst, config),				\
 		.adc_config = DT_INST_PROP(inst, adc_config),			\
 		.current_lsb = DT_INST_PROP(inst, current_lsb_microamps),	\
-		.rshunt = DT_INST_PROP(inst, rshunt_micro_ohms),			\
+		.rshunt = DT_INST_PROP(inst, rshunt_micro_ohms),		\
+		.cal = 8192U * (DT_INST_PROP(inst, current_lsb_microamps) *	\
+			DT_INST_PROP(inst, rshunt_micro_ohms) / 1000U) / 10000000U,\
 		.alert_config = DT_INST_PROP_OR(inst, alert_config, 0x01),	\
 		.alert_gpio = GPIO_DT_SPEC_INST_GET_OR(inst, alert_gpios, {0}),	\
 	};							    \
diff --git a/drivers/sensor/ina23x/ina237.h b/drivers/sensor/ina23x/ina237.h
index b028f5ffb9..e923bbaf05 100644
--- a/drivers/sensor/ina23x/ina237.h
+++ b/drivers/sensor/ina23x/ina237.h
@@ -50,6 +50,7 @@ struct ina237_config {
 	uint16_t adc_config;
 	uint32_t current_lsb;
 	uint32_t rshunt;
+	uint16_t cal;
 	const struct gpio_dt_spec alert_gpio;
 	uint16_t alert_config;
 };
```

I still noticed that it didn't pass clang formatting. Since this doesn't seem to be providing any extra value and I'm performing the correct order of operations any, I think I will leave the current PR the way it is.

Reply:
Likewise, using 64-bit math allows removing the possibility of overflow without compromising the precision of the result.

```c
#define INA237_CAL_SCALING 8192ULL
.cal = (INA237_CAL_SCALING * DT_INST_PROP(inst, current_lsb_microamps) * DT_INST_PROP(inst, rshunt_micro_ohms)) / 10000000000ULL;
```

Reply:
Ok, yep, makes sense. Added that and listed you as co-author.